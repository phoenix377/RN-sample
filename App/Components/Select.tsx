import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';

import {
  CONTAINER,
  PILL_LEFT,
  PILL_RIGHT,
  PILL_SELECTED,
  PILL_TEXT,
  PILL_TEXT_SELECTED,
} from './Select.styles';

type SelectProps = {
  value: 'imperial' | 'metric';
  onChange: (value: 'imperial' | 'metric') => void;
};

const Select: React.FC<SelectProps> = props => {
  return (
    <View style={CONTAINER}>
      <TouchableOpacity
        style={[PILL_LEFT, props.value === 'imperial' ? PILL_SELECTED : null]}
        onPress={() => props.onChange('imperial')}
      >
        <Text
          style={props.value === 'imperial' ? PILL_TEXT_SELECTED : PILL_TEXT}
        >
          Imperial
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={[PILL_RIGHT, props.value === 'metric' ? PILL_SELECTED : null]}
        onPress={() => props.onChange('metric')}
      >
        <Text style={props.value === 'metric' ? PILL_TEXT_SELECTED : PILL_TEXT}>
          Metric
        </Text>
      </TouchableOpacity>
    </View>
  );
};

export default Select;
