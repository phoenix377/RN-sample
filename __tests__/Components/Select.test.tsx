import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Select from '../../App/Components/Select';
import { TouchableOpacity } from 'react-native';

it('Select renders correctly', () => {
  const tree = renderer
    .create(<Select value="imperial" onChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Select calls onChange correctly', () => {
  const tree = renderer
    .create(<Select value="imperial" onChange={jest.fn(x => x)} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  const mockCallback = jest.fn();
  const wrapper = shallow(<Select value="metric" onChange={mockCallback} />);

  wrapper
    .find(TouchableOpacity)
    ?.first()
    ?.props()
    ?.onPress?.({} as any);

  wrapper.find(TouchableOpacity).first().simulate('press');
  expect(mockCallback.mock.calls[0][0]).toBe('imperial');
});
