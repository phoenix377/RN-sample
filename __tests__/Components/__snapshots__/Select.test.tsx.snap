// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select calls onChange correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4caf50",
        "borderBottomLeftRadius": 20,
        "borderTopLeftRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
        }
      }
    >
      Imperial
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e8f5e9",
        "borderBottomRightRadius": 20,
        "borderTopRightRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
        }
      }
    >
      Metric
    </Text>
  </View>
</View>
`;

exports[`Select renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4caf50",
        "borderBottomLeftRadius": 20,
        "borderTopLeftRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
        }
      }
    >
      Imperial
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e8f5e9",
        "borderBottomRightRadius": 20,
        "borderTopRightRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
        }
      }
    >
      Metric
    </Text>
  </View>
</View>
`;
